From 70d982b179a3d84c54277787a1defb35681c42de Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 14 Jan 2015 06:20:56 -0500
Subject: [infiniband] ocrdma: Fix ocrdma_query_qp() to report q_key value for UD QPs

Message-id: <83b2d0c62e21bf79807eb81b76352e7c85350442.1421216208.git.dledford@redhat.com>
Patchwork-id: 102845
O-Subject: [Patch RHEL7 2/2] RDMA/ocrdma: Fix ocrdma_query_qp() to report q_key value for UD QPs
Bugzilla: 1167256
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Mitesh Ahuja <mitesh.ahuja@emulex.com>

Signed-off-by: Mitesh Ahuja <mitesh.ahuja@emulex.com>
Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 95bf0093a95f864044d7782ddd15528350f915be)
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index ef6c78c..d8569ca 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -1410,6 +1410,8 @@ int ocrdma_query_qp(struct ib_qp *ibqp,
  mutex_unlock(&dev->dev_lock);
  if (status)
   goto mbx_err;
+ if (qp->qp_type == IB_QPT_UD)
+  qp_attr->qkey = params.qkey;
  qp_attr->qp_state = get_ibqp_state(IB_QPS_INIT);
  qp_attr->cur_qp_state = get_ibqp_state(IB_QPS_INIT);
  qp_attr->path_mtu =
-- 
1.7.1